docs/real-time/azure-signalr-scenario.md

```diff
@@ -1,21 +1,189 @@
+
+To use the emulator, chain a call to the <xref:Aspire.Hosting.AzureSignalRExtensions.RunAsEmulator(Aspire.Hosting.ApplicationModel.IResourceBuilder{Aspire.Hosting.ApplicationModel.AzureSignalRResource},System.Action{Aspire.Hosting.ApplicationModel.IResourceBuilder{Aspire.Hosting.Azure.AzureSignalREmulatorResource}})> method:
+
+```csharp
+using Aspire.Hosting.Azure;
+
+var builder = DistributedApplication.CreateBuilder(args);
+
+var signalR = builder.AddAzureSignalR("signalr", AzureSignalRServiceMode.Serverless)
+                     .RunAsEmulator();
+
+builder.AddProject<Projects.ApiService>("apiService")
+       .WithReference(signalR)
+       .WaitFor(signalR);
+
+// After adding all resources, run the app...
+```
+
+In the preceding example, the `RunAsEmulator` method configures the Azure SignalR Service resource to run as an emulator. The emulator is based on the `mcr.microsoft.com/signalr/signalr-emulator:latest` container image. The emulator is started when the app host is run, and it will be stopped when the app host is stopped.
+
+#### Azure SignalR Service modes
+
+While the Azure SignalR Service emulator only supports the _Serverless_ mode, the Azure SignalR Service resource can be configured to use either of the following modes:
+
+- <xref:Aspire.Hosting.ApplicationModel.AzureSignalRServiceMode.Default?displayProperty=nameWithType>
+- <xref:Aspire.Hosting.ApplicationModel.AzureSignalRServiceMode.Serverless?displayProperty=nameWithType>
+
+The _Default_ mode is the "default" configuration for Azure SignalR Service. Each mode has its own set of features and limitations. For more information, see [Azure SignalR Service modes](/azure/azure-signalr/concept-service-mode).
+
+> [!IMPORTANT]
+> The Azure SignalR Service emulator only works in _Serverless_ mode and the `AddNamedAzureSignalR` method doesn't support _Serverless_ mode.
+
+## Hub host integration
+
+There isn't an official .NET Aspire Azure SignalR [_client integration_](../fundamentals/integrations-overview.md#client-integrations). However, there is limited support for similar experiences. In these scenarios, the Azure SignalR Service acts as a proxy between the server (where the <xref:Microsoft.AspNetCore.SignalR.Hub> or <xref:Microsoft.AspNetCore.SignalR.Hub`1> are hosted) and the client (where the SignalR client is hosted). The Azure SignalR Service routes traffic between the server and client, allowing for real-time communication.
+
+> [!IMPORTANT]
+> It's important to disambiguate between .NET Aspire client integrations and the .NET SignalR client. SignalR exposes hubs—which act as a server-side concept—and SignalR clients connect to those hubs. The .NET projects that host SignalR hubs are where you integrate with .NET Aspire. The SignalR client is a separate library that connects to those hubs, in a different project.
```

Comment:
A cool diagram might help explain this complex design.

Reply:
I agree, what would that look like though? Something that shows the following?

```mermaid
graph TD;
    A(.NET Aspire App Host Project) --> B(Azure SignalR Service<br>'Hosting Integration');
    A --> C(ASP.NET Core Minimal API Project<br>'Client Integration');
    A --> D(Frontend Project <br> 'SignalR client');
```